pkg/resources/user_parameters.go

```diff
@@ -0,0 +1,399 @@
+		{Name: sdk.UserParameterWeekStart, Type: schema.TypeInt, Description: "Specifies the first day of the week (used by week-related date functions). `0`: Legacy Snowflake behavior is used (i.e. ISO-like semantics). `1` (Monday) to `7` (Sunday): All the week-related functions use weeks that start on the specified day of the week."},
+		{Name: sdk.UserParameterEnableUnredactedQuerySyntaxError, Type: schema.TypeBool, Description: "Controls whether query text is redacted if a SQL query fails due to a syntax or parsing error. If `FALSE`, the content of a failed query is redacted in the views, pages, and functions that provide a query history. Only users with a role that is granted or inherits the AUDIT privilege can set the ENABLE_UNREDACTED_QUERY_SYNTAX_ERROR parameter. When using the ALTER USER command to set the parameter to `TRUE` for a particular user, modify the user that you want to see the query text, not the user who executed the query (if those are different users)."},
+		{Name: sdk.UserParameterNetworkPolicy, Type: schema.TypeString, Description: "Specifies the network policy to enforce for your account. Network policies enable restricting access to your account based on users’ IP address. For more details, see [Controlling network traffic with network policies](https://docs.snowflake.com/en/user-guide/network-policies). Any existing network policy (created using [CREATE NETWORK POLICY](https://docs.snowflake.com/en/sql-reference/sql/create-network-policy))."},
+		{Name: sdk.UserParameterPreventUnloadToInternalStages, Type: schema.TypeBool, Description: "Specifies whether to prevent data unload operations to internal (Snowflake) stages using [COPY INTO <location>](https://docs.snowflake.com/en/sql-reference/sql/copy-into-location) statements."},
+	}
+
+	// TODO [SNOW-1348101][next PR]: extract this method after moving to SDK
+	for _, field := range userParameterFields {
+		fieldName := strings.ToLower(string(field.Name))
+
+		userParametersSchema[fieldName] = &schema.Schema{
+			Type:        field.Type,
+			Description: enrichWithReferenceToParameterDocs(field.Name, field.Description),
+			Computed:    true,
+			Optional:    true,
+			// TODO [SNOW-1348101][next PR]: uncomment and fill out
+			// ValidateDiagFunc: field.ValidateDiag,
+			// DiffSuppressFunc: field.DiffSuppress,
+		}
+	}
+}
+
+func userParametersProvider(ctx context.Context, d ResourceIdProvider, meta any) ([]*sdk.Parameter, error) {
+	client := meta.(*provider.Context).Client
+	id := helpers.DecodeSnowflakeID(d.Id()).(sdk.AccountObjectIdentifier)
+	userParameters, err := client.Parameters.ShowParameters(ctx, &sdk.ShowParametersOptions{
+		In: &sdk.ParametersIn{
+			User: id,
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+	return userParameters, nil
```

Comment:
```suggestion
	return client.Parameters.ShowParameters(ctx, &sdk.ShowParametersOptions{
		In: &sdk.ParametersIn{
			User: id,
		},
	})
```
BTW, I think these parameter providers could be in SDK - even as a `Users` method. WDYT?

Reply:
Yep, we could add them (however, for the generated objects we will have to modify the generator). I will add a TODO for the generator and I will add such method to Users, Databases, and Schemas.

Reply:
Done (for warehouses too)